Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -2039,6 +2057,11 @@ void FunctionDefinitionWriter::visit(AST::IndexAccessExpression& access)
 
 void FunctionDefinitionWriter::visit(AST::IdentifierExpression& identifier)
 {
+    auto it = m_overrides.find(identifier.identifier());
+    if (UNLIKELY(it != m_overrides.end())) {
+        serializeConstant(identifier.inferredType(), it->value);
```

Comment:
I had to apply the below patch for half otherwise we crash inside `std::get`

(inside `FunctionDefinitionWriter::serializeConstant` path for `half`)
```
@@ -2349,7 +2372,12 @@ void FunctionDefinitionWriter::serializeConstant(const Type* type, ConstantValue
             }
             case Primitive::F16: {
                 NumberToStringBuffer buffer;
-                WTF::numberToStringWithTrailingPoint(std::get<half>(value), buffer);
+                ASSERT(halfPtr);
+                if (halfPtr)
+                    WTF::numberToStringWithTrailingPoint(*halfPtr, buffer);
+                else
+                    WTF::numberToStringWithTrailingPoint(std::get<float>(value), buffer);
                 m_stringBuilder.append(&buffer[0], "h");
                 break;
             }
```